### PR TITLE
Porting `pom-operator` from Kotlin to JAVA

### PR DESCRIPTION
**Pull request by carlosu7**
# Description
Porting all classes and functions inside `io.github.pixee.maven.operator` from Kotlin to JAVA.

I didn't port any tests in order to assert that all JAVA classes work as expected and no assertions were altered by mistake.

# Checklist:

- [ ] I did my own review
- [ ] Commented Code - particularly in hard sections
- [ ] Docs (Javadocs)
- [ ] No TODOs / Issues on Code
- [ ] Tests: Unit Testing
- [ ] Tests: Assertions Look Good and Sane
- [ ] Tests: Coverage
- [ ] Integration Testing
- [ ] Java Test Fixtures
- [ ] Security Risks Considered and Addressed / Mitigated

### CODE REVIEW

**Comment by carlosu7** [comment]
@aldrinleal 
@nahsra 

Could you help me review this PR?

File: pom.xml [hunk 1]
```diff
@@ -50,8 +50,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
```
Comment (aldrinleal) [inline pom.xml]: I wonder if you could leverage `maven.compiler.release`.

https://stackoverflow.com/questions/59049980/maven-compiler-release-as-an-replacement-for-source-and-target

File: pom.xml [hunk 2]
```diff
@@ -238,6 +238,17 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.30</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
```
Comment (aldrinleal) [inline pom.xml]: Did we also consider Google's Guava? (Just wondering, not a blocker by any means)

Nice to see Lombok, although I believe it hinders a bit bc it becomes heavily IDE-dependent, thus rendering moot the whole point of porting towards Java

Comment (aldrinleal) [reply pom.xml]: Also: We are only using two methods from `commons-collections` it seems. Can't we perhaps inline? TBH, it looks like an Idiom we could evolve further into the `Util` class wdyt?

Comment (aldrinleal) [reply pom.xml]: lombok-wise, why not leverage commons-lang`s builders classes instead?

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html

**Review by aldrinleal — CHANGES_REQUESTED** [review]
Going to ask three things:

a. Getting rid of Lombok and Commons-collections (no changes in `.github/` and `pom.xml` would be a good goal tbh)
b. Making the build actually work - currently its failing due to actually not using `maven.compiler.release` - We use [act](https://github.com/nektos/act) for that e.g.:

```
$ act -j build
...
 [INFO] Changes detected - recompiling the module!
| [INFO] Compiling 42 source files to /home/aldrin/projetos/sources/pom-operator/target/classes
| [INFO] ------------------------------------------------------------------------
| [INFO] BUILD FAILURE
| [INFO] ------------------------------------------------------------------------
| [INFO] Total time:  7.793 s
| [INFO] Finished at: 2023-10-10T14:23:37Z
| [INFO] ------------------------------------------------------------------------
| [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project pom-operator: Fatal error compiling: invalid target release: 17 -> [Help 1]
| [ERROR] 
```

(thats actually where the Integration Test will kick in)

c. I'm not seeing coverage information. Once the past two items are solved third one should be good to watch

Thanks

**Comment by carlosu7** [comment]
@aldrinleal I have covered point a and b. For point c (coverage information) I'm not where I can watch it.

Captura de pantalla 2023-10-10 a la(s) 14 59 43

**Review by aldrinleal — APPROVED** [review]
LGTM

**Comment by aldrinleal** [comment]
Be wary that coverage went down a little bit (its on the end of the `act -j build` command)

```
[Java CI/build-1] ⭐ Run Main Log coverage percentage
[Java CI/build-1]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/7] user= workdir=
| coverage = 0.8823443650677941
| branch coverage = 0.7523219814241486
```

File: src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml
Comment (aldrinleal) [inline src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml:1]: I'd also check why this extra `>` happened and confirm.

File: src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml [hunk 3]
```diff
@@ -1788,7 +1788,7 @@
                 <phase>generate-sources</phase>
                 <configuration>
                   <target>
-                    <mkdir dir="${project.build.directory}/dependency"></mkdir>
+                    <mkdir dir="${project.build.directory}/dependency"></mkdir>>
```
Comment (aldrinleal) [inline src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml]: I'd also check why this extra `>` happened.

Comment (carlosu7) [reply src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml]: @aldrinleal done

Comment (aldrinleal) [reply src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml]: Awesome. btw I already approved it. Do you want me to tag and release @carlosu7? (cc @nahsra )

Comment (carlosu7) [reply src/test/resources/io/github/pixee/maven/operator/test/pom-hack23-cia-result.xml]: @aldrinleal we're going to bring this java source code into the codemodder repository, I have a draft [here](https://github.com/carlosu7/codemodder-java/pull/1) I just need to update it with the last 2 commits from here and also change the destination. When I do that, I will open a PR for it. (cc @nahsra )

**Pull request closed without merging**